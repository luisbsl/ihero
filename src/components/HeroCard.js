import React from 'react'
import { createFragmentContainer } from 'react-relay'
import { View, Image, Text, Dimensions, TouchableHighlight } from 'react-native'
import styled from 'styled-components'

import HeroStatsBadge from './HeroStatsBadge'

const Container = styled.View`
  flex-direction: row;
  margin-bottom: 10px;
  border-color: #bfbfbf;
  border-width: 2px;
  align-items: center;
  width: ${(Dimensions.get('window').width - 10)}px;
  background-color: #e6e6e6;
`

const ImageContainer = styled.View`
  align-items: center;
  width: 30%;
`

const InfoWrapper = styled.View`
  flex: 1;
  height: 100px;
`

const InfoContainer = styled.View`
  height: 80%; 
  padding-top: 25px;
`

const StyledImage = styled.Image`
  margin-bottom: 10px;
  margin-top: 10px;
  width: 90px;
  height: 90px;
  border-radius: 100;
`

const Info = styled.Text`
  color: #4d4d4d; 
  font-weight: bold;
`
const BadgesContainer = styled.View`
  flex: 1;
  flex-direction: row; 
  margin-top: 7px;
`

class HeroCard extends React.Component {
  render() {
    const { navigate } = this.props.navigation;
    return (
      this.props.hero
        ?
        <TouchableHighlight onPress={() => navigate('HeroDetail', { hero: this.props.hero })}>
          <Container>
            <ImageContainer>
              <StyledImage
                resizeMode={'cover'}
                source={{ uri: `${this.props.hero.image}` }} />
            </ImageContainer>
            <InfoWrapper>
              <InfoContainer>
                <Info>{this.props.hero.name.toUpperCase()}</Info>

                <BadgesContainer>
                  <HeroStatsBadge bgColor={'#0066ff'} statName={'Comics'} statNumber={this.props.hero.comics} />
                  <View style={{marginRight: 5}} />
                  <HeroStatsBadge bgColor={'#00cc00'} statName={'Series'} statNumber={this.props.hero.series} />
                </BadgesContainer>
              </InfoContainer>
            </InfoWrapper>
          </Container>
        </TouchableHighlight>
        : null
      // <ErrorCard error='Error! Hero properties is required!' />
    )
  }
}

export default createFragmentContainer(HeroCard, graphql`
  fragment HeroCard_hero on Hero { 
    id
    name
    description
    image
    comics
    series
    stories
  }
`)