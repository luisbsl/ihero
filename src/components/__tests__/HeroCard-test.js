import React from 'react'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import HeroCard from '../HeroCard'

test('renders correctly', () => {
  const hero = {
    "id": "1009187",
    "image": "http://i.annihil.us/u/prod/marvel/i/mg/6/60/5261a80a67e7d/standard_fantastic.jpg",
    "name": "Black Panther",
    "description": "",
    "comics": 543,
    "series": 175,
    "stories": 745
  }
  const navigation = {
    navigate: {}
  }
  const tree = renderer.create(<HeroCard hero={hero} navigation={navigation} />).toJSON()
  expect(tree).toMatchSnapshot()
})