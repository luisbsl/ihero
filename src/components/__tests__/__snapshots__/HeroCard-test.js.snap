// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#e6e6e6",
          "borderColor": "#bfbfbf",
          "borderWidth": 2,
          "flexDirection": "row",
          "marginBottom": 10,
          "width": 740,
        },
        null,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "width": "30%",
          },
          undefined,
        ]
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "http://i.annihil.us/u/prod/marvel/i/mg/6/60/5261a80a67e7d/standard_fantastic.jpg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 100,
              "height": 90,
              "marginBottom": 10,
              "marginTop": 10,
              "width": 90,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 100,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": "80%",
              "paddingTop": 25,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#4d4d4d",
                "fontWeight": "bold",
              },
              undefined,
            ]
          }
        >
          BLACK PANTHER
        </Text>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginTop": 7,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#0066ff",
                "borderRadius": 2,
                "height": 20,
                "width": 100,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "width": 40,
                    },
                    Object {
                      "width": 40,
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  543
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 3,
                      "width": 90,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                      },
                      undefined,
                    ]
                  }
                >
                  Comics
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#00cc00",
                "borderRadius": 2,
                "height": 20,
                "width": 100,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "width": 40,
                    },
                    Object {
                      "width": 40,
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  175
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 3,
                      "width": 90,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                      },
                      undefined,
                    ]
                  }
                >
                  Series
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
